pkg/codegen/hcl2/model/type_object.go

```diff
@@ -80,6 +82,32 @@ func (t *ObjectType) Traverse(traverser hcl.Traverser) (Traversable, hcl.Diagnos
 	return propertyType, nil
 }
 
+func (t *ObjectType) hash(stack objTypeSet) uint32 {
+	if t.h == 0 {
+		if stack.add(t) {
+			return hashKindObject
+		}
+		defer stack.delete(t)
+
+		fnv := newFNV()
+		fnv.addUint32(hashKindObject)
+
+		keys := make([]string, 0, len(t.Properties))
+		for k := range t.Properties {
+			keys = append(keys, k)
+		}
+		sort.Strings(keys)
+
+		for _, k := range keys {
```

Comment:
I don't know if we ever have a thing with super-fat objects, but you can consider giving up hashing past 8th field or such.